src/ethereum/EthereumEngine.ts

```diff
@@ -1302,11 +1304,13 @@ export class EthereumEngine extends CurrencyEngine<
       return null
     }
     // Other params checks:
-    // The transaction must have a known nonce used and no data payload.
     if (
       replacedTxOtherParams == null ||
+      // The transaction must have a known nonce used.
       replacedTxOtherParams.nonceUsed == null ||
-      replacedTxOtherParams.data != null
+      // Must not have data payload unless it's normalized.
+      (replacedTxOtherParams.data != null &&
```

Comment:
Should just be `!replacedTxOtherParams.isFromMakeSpend`. It's dangerous to check the `.data` as it might be null if we got it off the network even if it's a contract tx